atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -259,7 +260,11 @@ public void compactInternally() {
     }
 
     private String getPrimaryKeyConstraintName(String tableName) {
-        String primaryKeyConstraintPrefix = "pk_";
-        return primaryKeyConstraintPrefix + tableName.substring(primaryKeyConstraintPrefix.length());
+        return truncateToMaxOracleLength("pk_" + tableName);
+    }
+
+    private String truncateToMaxOracleLength(String constraintName) {
+        return constraintName
+                .substring(0, Math.max(OracleTableNameMapper.ORACLE_MAX_TABLE_NAME_LENGTH, constraintName.length()));
```

Comment:
Looks to be a duplicate of the one in the AbstractDbWriteTable. I'll admit I don't have that much context on this, but it seems a bit concerning to me that this looks like it happens without any logging/feedback to users.

Also this is probably worth a unit test.